server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -270,7 +285,12 @@ private VersionValue getUnderLock(final BytesRef uid, Maps currentMaps) {
             return value;
         }
 
-        return tombstones.get(uid);
+        value = tombstones.get(uid);
+        if (value != null) {
+            return value;
+        }
+
+        return archiver.get(uid);
```

Comment:
I think it is OK initially, but we may need to build a mechanism in `IndexingMemoryController` to clear out the archive (making it "unsafe") or to flush rather than refresh when the archive has too much memory use.

Reply:
I've created ES-5921.